scipy/linalg/tests/test_matfuncs.py

```diff
@@ -1045,3 +1048,13 @@ def test_empty(self):
         b = np.empty((5, 0))
         res = khatri_rao(a, b)
         assert_allclose(res, np.empty((15, 0)))
+
+@pytest.mark.parametrize('func', [cosm, sinm, tanm, coshm, sinhm, tanhm])
+@pytest.mark.parametrize('dt', [int, float, np.float32, complex, np.complex64])
```

Comment:
These are the dtypes tested in gh-20295, so that looks fine. And I would use `dtype` as the name, but I see gh-20295 uses `dt`, so this can, too.